apps/api/src/models/post.ts

```diff
@@ -0,0 +1,45 @@
+import mongoose, { Document, Schema } from 'mongoose';
+
+interface ICategory {
+  name: string;
+}
+
+interface IComment {
+  author: string;
+  content: string;
+}
+interface IPost extends Document {
+  title: string;
+  image?: string;
+  description: string;
+  category: ICategory;
+  comments?: Array<IComment>;
+}
+
+export const postSchema = new Schema<IPost>(
+  {
+    title: {
+      type: String,
+      required: [true, 'Property is required']
+    },
+    image: {
+      type: String
+    },
+    description: {
+      type: String,
+      required: [true, 'Property is required']
+    },
+    category: {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: 'Category'
+    },
+    comments: {
```

Comment:
This is a correct approach, but could have performance issues once we have a lot of comments because this will nest the document inside another document. A better approach can be to use an `ObjectId` as a reference to a different model same as you did for category, see the example below

```
comments: [{
  type: mongoose.Schema.Types.ObjectId,
  ref: 'Comment' // Comment will be a new model with author and content as the properties
}]
```